client.go

```diff
@@ -186,14 +186,9 @@ func (c *Client) execEnqueue(ctx context.Context, jobs []*Job, q adapter.Queryab
 //
 // After the Job has been worked, you must call either Job.Done() or Job.Error() on it
 // in order to commit transaction to persist Job changes (remove or update it).
-func (c *Client) LockJob(ctx context.Context, queue string) (*Job, error) {
-	sql := `SELECT job_id, queue, priority, run_at, job_type, args, error_count, last_error, created_at
-FROM gue_jobs
-WHERE queue = $1 AND run_at <= $2
-ORDER BY priority ASC
-LIMIT 1 FOR UPDATE SKIP LOCKED`
-
-	return c.execLockJob(ctx, true, sql, queue, time.Now().UTC())
+func (c *Client) LockJob(ctx context.Context, queue string, jobTypes ...string) (*Job, error) {
+	sql, args := newLockJobQuery(queue, jobTypes)
```

Comment:
this is nice from the DRY point of view, but makes code readability way worse

I prefer having full queries here

Reply:
The problem is that now the job_type condition is not fixed as the previous conditions where before.
So to keep it there we need:
- Define the first part of the SQL
- Define the base Where clause
- Define the base args
- If there are job types, append to the where clause the condition and append to args the array
- Add the end of the query
- Call the exec
As you can see it gets messy and a ton of duplicated lines among functions.

Could you suggest another approach that would not make those functions messy?

PS: I've been always trying to separate as much as possible logic from SQL, as you separate HTML/CSS from logic.

Reply:
> Could you suggest another approach that would not make those functions messy?

```
sql := `SELECT ... `
if len(jobTypes) > 0 {
  sql = `SELECT ...`
}
```

and tests that will cover branching